### PR TITLE
Wall refactor

### PR DESCRIPTION
**Pull request by sqreek**
[SUS-298](https://wikia-inc.atlassian.net/browse/SUS-298)

This Pr contains Wall refactor.
@Wikia/sustaining-team

### CODE REVIEW

**Comment by sqreek** [comment]
ready for CR 
ping @macbre

File: extensions/wikia/Forum/tests/ForumTest.php [hunk 1]
```diff
@@ -54,8 +54,17 @@ private function getTitleMock( $key ) {
 		return $titleMock;
 	}
 
-	private function getBoardMock() {
-		$boardMock = $this->getMock( 'ForumBoard', [ 'getBoardInfo', 'getDescriptionWithoutTemplates', 'getTitle' ] );
+	/**
+	 * @param $title Title
+	 * @return PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function getBoardMock( $title ) {
```
Comment (macbre) [inline extensions/wikia/Forum/tests/ForumTest.php]: I'd specify the instance of `$title` argument here - `Title $title`

File: extensions/wikia/Forum/ForumBoard.class.php [hunk 2]
```diff
@@ -100,7 +102,7 @@ private function getBoardInfoFromMaster() {
 		$forumBoardInfo->setName( $this->getTitle()->getText() );
 		$forumBoardInfo->setUrl( $this->getTitle()->getFullURL() );
 		$forumBoardInfo->setDescription( $this->getDescriptionWithoutTemplates() );
-		$memKey = wfMemcKey( 'forum_board_info', $this->getId() );
+		$memKey = wfMemcKey( self::FORUM_BOARD_INFO, self::CACHE_VERSION , $this->getId() );
```
Comment (macbre) [inline extensions/wikia/Forum/ForumBoard.class.php]: A helper method is still better. Here you can still omit one instance of memcache key generation - see 4671dc4 diff

File: extensions/wikia/Forum/ForumBoard.class.php [hunk 3]
```diff
@@ -135,4 +134,11 @@ private function getLastPost( $rev_id ) {
 		}
 		return $postInfo;
 	}
+
+	/**
+	 * @return string cache key for getBoardInfo()
+	 */
+	private static function getBoardInfoCacheKey() {
```
Comment (macbre) [inline extensions/wikia/Forum/ForumBoard.class.php]: Let's pass `$this->getId()`as an argument here. And we're done :)

File: extensions/wikia/Forum/ForumBoard.class.php [hunk 4]
```diff
@@ -133,4 +134,12 @@ private function getLastPost( $rev_id ) {
 		}
 		return $postInfo;
 	}
+
+	/**
+	 * @param $forumBoardId integer
+	 * @return string cache key for getBoardInfo()
+	 */
+	private static function getBoardInfoCacheKey( $forumBoardId ) {
+		return implode( ':', [ self::FORUM_BOARD_INFO, self::CACHE_VERSION, $forumBoardId ] );
```
Comment (macbre) [inline extensions/wikia/Forum/ForumBoard.class.php]: Let's call `wfMemcKey` inside this method and return a full key to be used (including per wiki prefix) from this helper.

**Pull request merged**